src/Sema.zig

```diff
@@ -9618,6 +9618,15 @@ fn funcCommon(
         {
             return sema.fail(block, param_src, "non-pointer parameter declared noalias", .{});
         }
+
+        if (cc_resolved == .Interrupt) {
+            const err_code_size = target.ptrBitWidth();
+            switch (i) {
+                0 => if (param_ty.zigTypeTag(mod) != .Pointer) return sema.fail(block, param_src, "parameter must be a pointer type", .{}),
+                1 => if (param_ty.bitSize(mod) != err_code_size) return sema.fail(block, param_src, "parameter must have a bit size of {d}", .{err_code_size}),
+                else => return sema.fail(block, param_src, "Interrupt calling convention supports up to 2 parameters, found {d}", .{i + 1}),
+            }
+        }
```

Comment:
Almost there, just need to handle other architectures too.
```suggestion
        if (cc_resolved == .Interrupt) switch (target.cpu.arch) {
            .x86, .x86_64 => {
                const err_code_size = target.ptrBitWidth();
                switch (i) {
                    0 => if (param_ty.zigTypeTag(mod) != .Pointer) return sema.fail(block, param_src, "parameter must be a pointer type", .{}),
                    1 => if (param_ty.bitSize(mod) != err_code_size) return sema.fail(block, param_src, "parameter must have a bit size of {d}", .{err_code_size}),
                    else => return sema.fail(block, param_src, "Interrupt calling convention supports up to 2 parameters, found {d}", .{i + 1}),
                }
            },
            else => return sema.fail(block, param_src, "parameters are not allowed with Interrupt calling convention", .{}),
        };
```